common/src/main/java/de/michelinside/glucodatahandler/common/receiver/NotificationReceiver.kt

```diff
@@ -0,0 +1,160 @@
+package de.michelinside.glucodatahandler.common.receiver
+
+import android.os.Bundle
+import android.service.notification.NotificationListenerService
+import android.service.notification.StatusBarNotification
+import android.util.Log
+import android.view.View
+import android.view.ViewGroup
+import android.widget.RemoteViews
+import android.widget.TextView
+import de.michelinside.glucodatahandler.common.ReceiveData
+import de.michelinside.glucodatahandler.common.SourceState
+import de.michelinside.glucodatahandler.common.SourceStateData
+import de.michelinside.glucodatahandler.common.notifier.DataSource
+import de.michelinside.glucodatahandler.common.utils.GlucoDataUtils
+
+class NotificationReceiver : NotificationListenerService(), NamedReceiver {
+    private val LOG_ID = "GDH.NotificationReceiver"
+
+    override fun getName(): String {
+        return LOG_ID
+    }
+
+    override fun onNotificationPosted(statusBarNotification: StatusBarNotification?) {
+        if (isRegistered()) {
+            statusBarNotification?.let { sbn ->
+                if (sbn.packageName == "com.dexcom.g7") {
+                    Log.i(LOG_ID, "handling dexcom g7 notification")
+                    val notification = sbn.notification
+                    val contentView = notification.contentView
+                    val bigContentView = notification.bigContentView
+
+                    // Try processing different RemoteViews
+                    if (processRemoteViews(contentView)) return
+                    if (processRemoteViews(bigContentView)) return
+
+                    // If we get here, we couldn't find the glucose value
+                    Log.d(LOG_ID, "Could not find glucose value in notification")
+                }
+                if (sbn.packageName == "com.insulet.myblue.pdm") {
```

Comment:
Do you have any screenshot? Makes it easier for me to understand.
Is there also the glucose value present?
At the moment, 2 notifications are parsed, I would prefer only one for one app. 
But I do not have the knowledge for all these systems...